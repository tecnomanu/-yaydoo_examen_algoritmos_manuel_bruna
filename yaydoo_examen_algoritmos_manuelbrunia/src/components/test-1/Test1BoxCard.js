import React from 'react'

export const Test1BoxCard = ({ name, state }) => {
    return (
        <div className="card">
            <div className="card-header">
                {name}
            </div>
            <ul className="list-group list-group-flush">
                <li className="list-group-item">Claridad del problema: {state[0]}</li>
                <li className="list-group-item">Originalidad: {state[1]}</li>
                <li className="list-group-item">Dificultad: {state[2]}</li>
            </ul>
        </div>
    )
}
