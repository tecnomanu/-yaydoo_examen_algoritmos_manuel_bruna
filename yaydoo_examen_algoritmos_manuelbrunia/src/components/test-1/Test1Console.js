import React from 'react'

export const Test1Console = ({alice, bob, results}) => {
    return (
        <>
            <strong>Console:</strong><br/>
            <code>const a: [{alice.map((str,idx) => { return(`${str}${(idx < alice.length-1 ? ', ' : '')}` ) })}];</code><br/>
            <code>const b: [{bob.map((str,idx) => { return(`${str}${(idx < bob.length-1 ? ', ' : '')}` ) })}];</code><br/>
            <code>result: {results[0]}, {results[1]}</code>
        </>
    )
}
