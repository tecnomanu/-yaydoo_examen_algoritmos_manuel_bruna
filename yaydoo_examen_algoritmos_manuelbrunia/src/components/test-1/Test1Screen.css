.background-code{
    background-color: #ccc;
}