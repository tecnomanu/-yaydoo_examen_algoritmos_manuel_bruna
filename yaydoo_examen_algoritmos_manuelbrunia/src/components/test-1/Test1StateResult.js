import React from 'react'

export const Test1StateResult = ( {resultAlice, resultBob} ) => {
    return (
        <div>
            <h1>{resultAlice} vs {resultBob}</h1>
            <h2>{
                (resultAlice === resultBob) 
                ? 'Hay empate'
                : (
                    (resultAlice > resultBob) 
                    ? 'Gana Alice'
                    : 'Gana Bob'
                )
                    
            }</h2>
        </div>
    )
}
