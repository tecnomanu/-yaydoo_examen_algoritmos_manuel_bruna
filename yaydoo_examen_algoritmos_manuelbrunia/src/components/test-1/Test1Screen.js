import React, { useState } from 'react'
import { randomTreeFloorNumber } from '../../helpers/randomTreeFloorNumber';
import { solveMatch } from '../../helpers/solveMatch';
import { Test1BoxCard } from './Test1BoxCard';
import { Test1Console } from './Test1Console';
import './Test1Screen.css';
import { Test1StateResult } from './Test1StateResult';

export const Test1Screen = () => {
    const min = 1;
    const max = 100;
    const [alice_state, setAliceState] = useState([17,28,30]);
    const [bob_state, setBobState] = useState([99,16,8]);

    const [result_state, setResultState] = useState(solveMatch(alice_state, bob_state));

    const handleNumbersChange = () => {
        setAliceState(a => randomTreeFloorNumber(min, max))
        setBobState(b => randomTreeFloorNumber(min, max))
        setResultState(r => solveMatch(alice_state, bob_state));
    }

    return (
        <div className="container-fluid">
            <div className="row">
                <div className="col-md-8 pt-2 pb-2">
                    <h1 className="text-center">Prueba 1 - Alice vs Bob</h1>
                    <h5 className="text-center">Alice y Bob se arman en contienda para ver quien gana segun sus aptitudes.</h5>
                    <div className="row justify-content-center">
                        <div className="col-md-6">
                            <Test1BoxCard state={alice_state} name={'Alice'}></Test1BoxCard>
                        </div>
                        <div className="col-md-6">
                            <Test1BoxCard state={bob_state} name={'Bob'}></Test1BoxCard>
                        </div>
                    </div>
                    <div className="row justify-content-center text-center">
                        <div className="col-md-4 text-center">
                            <Test1StateResult resultAlice={result_state[0]} resultBob={result_state[1]}></Test1StateResult>
                            <button type="button" className="btn btn-success" onClick={handleNumbersChange}>Probar otro numero</button>
                        </div>
                    </div>
                </div>
                <div className="col-md-4 background-code pt-4 pb-4">
                    <Test1Console alice={alice_state} bob={bob_state} results={result_state}></Test1Console>
                </div>
            </div>
        </div>
    )
}
