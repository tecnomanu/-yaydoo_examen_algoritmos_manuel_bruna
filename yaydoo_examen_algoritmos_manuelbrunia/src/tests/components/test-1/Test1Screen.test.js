import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { Test1Screen } from '../../../components/test-1/Test1Screen';

describe("Test on <Test1Screen/>", ()=>{

    test('SnapShot of <Test1Screen/>', ()=>{
        const wrapper = shallow(<Test1Screen />);
        expect(wrapper).toMatchSnapshot();
    })

})