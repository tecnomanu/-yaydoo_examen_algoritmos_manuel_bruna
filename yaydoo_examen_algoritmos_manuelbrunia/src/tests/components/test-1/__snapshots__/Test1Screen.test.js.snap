// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Test1Screen/> SnapShot of <Test1Screen/> 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-8 pt-2 pb-2"
    >
      <h1
        className="text-center"
      >
        Prueba 1 - Alice vs Bob
      </h1>
      <h5
        className="text-center"
      >
        Alice y Bob se arman en contienda para ver quien gana segun sus aptitudes.
      </h5>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-md-6"
        >
          <Test1BoxCard
            name="Alice"
            state={
              Array [
                17,
                28,
                30,
              ]
            }
          />
        </div>
        <div
          className="col-md-6"
        >
          <Test1BoxCard
            name="Bob"
            state={
              Array [
                99,
                16,
                8,
              ]
            }
          />
        </div>
      </div>
      <div
        className="row justify-content-center text-center"
      >
        <div
          className="col-md-4 text-center"
        >
          <Test1StateResult
            resultAlice={2}
            resultBob={1}
          />
          <button
            className="btn btn-success"
            onClick={[Function]}
            type="button"
          >
            Probar otro numero
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-4 background-code pt-4 pb-4"
    >
      <Test1Console
        alice={
          Array [
            17,
            28,
            30,
          ]
        }
        bob={
          Array [
            99,
            16,
            8,
          ]
        }
        results={
          Array [
            2,
            1,
          ]
        }
      />
    </div>
  </div>
</div>
`;
